x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/accesscontrol/FieldExtractor.java

```diff
@@ -76,8 +75,6 @@ static void extractFields(Query query, Set<String> fields) throws UnsupportedOpe
             fields.add(pointInSetQuery.getField());
         } else if (query instanceof FieldExistsQuery fieldExistsQuery) {
             fields.add(fieldExistsQuery.getField());
-        } else if (query instanceof DocValuesNumbersQuery docValuesNumbersQuery) {
-            fields.add(docValuesNumbersQuery.getField());
```

Comment:
I am not entirely sure what consequences this has. I do not see a way to retrieve the field from the corresponding query that is generated now in place of the former DocValuesNumbersQuery.